packages/salesforcedx-vscode-core/src/commands/forceOrgOpen.ts

```diff
@@ -6,24 +6,66 @@
  */
 
 import {
+  CliCommandExecutor,
   Command,
   SfdxCommandBuilder
 } from '@salesforce/salesforcedx-utils-vscode/out/src/cli';
+import { ContinueResponse } from '@salesforce/salesforcedx-utils-vscode/out/src/types';
+import * as vscode from 'vscode';
 import { nls } from '../messages';
+import { telemetryService } from '../telemetry';
+import { getRootWorkspacePath } from '../util';
 import {
   EmptyParametersGatherer,
   SfdxCommandlet,
   SfdxCommandletExecutor,
   SfdxWorkspaceChecker
 } from './commands';
 
+interface CLIOrgData {
+  status: number;
+  result: {
+    orgId: string;
+    url: string;
+    username: string;
+  };
+}
+
 class ForceOrgOpenExecutor extends SfdxCommandletExecutor<{}> {
+  protected showChannelOutput = true;
   public build(data: {}): Command {
-    return new SfdxCommandBuilder()
+    const builder = new SfdxCommandBuilder()
       .withDescription(nls.localize('force_org_open_default_scratch_org_text'))
       .withArg('force:org:open')
-      .withLogName('force_org_open_default_scratch_org')
-      .build();
+      .withLogName('force_org_open_default_scratch_org');
+    if (process.env.SFDX_CONTAINER_MODE) {
+      builder.withArg('--urlonly').withJson();
+    }
+    return builder.build();
+  }
+
+  public execute(response: ContinueResponse<string>): void {
+    const cancellationTokenSource = new vscode.CancellationTokenSource();
+    const cancellationToken = cancellationTokenSource.token;
+
+    const execution = new CliCommandExecutor(this.build(response.data), {
+      cwd: getRootWorkspacePath()
+    }).execute(cancellationToken);
+
+    if (process.env.SFDX_CONTAINER_MODE) {
+      execution.stdoutSubject.subscribe(cliResponseJSON => {
+        try {
+          const cliOrgData: CLIOrgData = JSON.parse(cliResponseJSON.toString());
+          const authenticatedOrgUrl: string = cliOrgData.result.url;
+          if (authenticatedOrgUrl) {
+            vscode.env.openExternal(vscode.Uri.parse(authenticatedOrgUrl));
+          }
+        } catch (e) {
+          telemetryService.sendErrorEvent(e.message, e.stack);
+        }
+      });
+    }
+    this.attachExecution(execution, cancellationTokenSource, cancellationToken);
   }
```

Comment:
I needed to call attachExecution so that these methods are called to show the CLI output. However, because of the `--json` flag added to the open command, the output is in json format and not quite as UI friendly as when the output is wrapped in a message.
`channelService.streamCommandOutput(execution);`

   `if (this.showChannelOutput) {
      channelService.showChannelOutput();
    }`